app/components/work_packages/types/pattern_autocompleter.html.erb

```diff
@@ -0,0 +1,119 @@
+<%#-- copyright
+OpenProject is an open source project management software.
+Copyright (C) the OpenProject GmbH
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License version 3.
+
+OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+Copyright (C) 2006-2013 Jean-Philippe Lang
+Copyright (C) 2010-2013 the ChiliProject Team
+
+This program is free software; you can redistribute it and/or
+modify it under the terms of the GNU General Public License
+as published by the Free Software Foundation; either version 2
+of the License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+
+See COPYRIGHT and LICENSE files for more details.
+
+++#%>
+
+<%=
+  content_tag(
+    :div,
+    class: "pattern-autocompleter",
+    "data-controller": "pattern-autocompleter",
+    "data-pattern-autocompleter-pattern-initial-value": @value
+  ) do
+%>
+  <%=
+    @input.builder.hidden_field(
+      name,
+      value: @value,
+      data: {
+        "pattern-autocompleter-target": "formInput"
+      }
+    )
+  %>
+
+  <template data-pattern-autocompleter-target="tokenTemplate">
+    <%=
+      render(
+        Primer::Beta::Label.new(
+          contenteditable: false,
+          tag: :span, display: :inline, scheme: :accent,
+          "data-role": :token
+        )
+      ) do
+    %>
+      <%= content_tag(:span, style: "cursor: default;", "data-role": "token-text") { "__VALUE__" } %>
+      <%=
+        render(
+          Primer::Beta::Octicon.new(
+            icon: :x, size: :xsmall, style: "cursor: pointer;margin: 0.2em;",
+            "data-action": "click->pattern-autocompleter#remove_token"
+          )
+        )
+      %>
+    <% end %>
+  </template>
+
+  <%= content_tag(:div, style: "position: relative;") do %>
+      <%=
+        render(
+          Primer::Beta::Octicon.new(
+            icon: "triangle-down",
+            size: :small,
+            style: "cursor: pointer;position: absolute;right: 1em;top: 0.5em;",
```

Comment:
Position absolute has some drawbacks in this case. You are enforcing it to lay at that specific place no matter what the content of the input is. Thus the content and the arrow can overlap:

<img width="485" alt="Bildschirmfoto 2025-01-15 um 08 18 12" src="https://github.com/user-attachments/assets/0af2227a-01a3-46c6-9337-ec330deb9f70" />

Reply:
I didn't see that arrow down at all until now :see_no_evil:

Devils advocate: Do we need this arrow at all? It makes the component look like a select, when it is mostly like a text input. E.g. we'd also not show an arrow down in a text input that allows autocompleting emojis (which I think is a good analogy for what we are doing here).